woocommerce/wc-admin
# Reviews: Create wp.data dataStore

Fixes https://github.com/woocommerce/woocommerce-admin/issues/3376

Replace wc-api usage of reviews data with wp.data implementation. The reducer and selectors are made to replicate the same API as wc-api.

### Detailed test instructions:

1. Make sure you have an unread review.
2. Load the homescreen and see an alert dot on the Reviews header tab.
3. Click the Reviews header tab and see the reviews populate.
4. Coerce an error by adding this function to `src/API/ProductReviews.php`

```php
public function get_items() {
    return new WP_Error( 'Take me to flavortown` );
}
```
5. See the Reviews header tab display an error.

### Note

Using the headers to get a total count is an often used pattern, ie `response.headers.get( 'x-wp-total' )`. This PR introduces a control `fetchWithHeaders` which can ultimately be pulled out from the reviews folder and somewhere reusable.

### Changelog Note:

- dev: Replace Review's fresh-data with wp.data data store.
